geoportailv3/static/js/profile/profiledirective.js

```diff
@@ -181,6 +206,27 @@ app.ProfileController = function($scope) {
 
 
   this['point'] = null;
+
+  $scope.$watch(goog.bind(function() {
+    return this['profileData'];
+  }, this), goog.bind(function(newVal, oldVal) {
+    if (goog.isDef(newVal)) {
+      var i;
+      var len = newVal.length;
+      var lineString = new ol.geom.LineString([],
+          /** @type {ol.geom.GeometryLayout} */ ('XYM'));
+      for (i = 0; i < len; i++) {
+        var p = newVal[i];
+        p = new ol.geom.Point([p['x'], p['y']]);
+        p = /** @type {ol.geom.Point} */
+            (p.transform(
+                'EPSG:2169', this['map'].getView().getProjection()));
```

Comment:
The `transform` method transform "in-place", so you can avoid the type-cast by ignoring the return value:

```js
  p = new ol.geom.Point(p['x'], p['y']);
  p.transform('EPSG:2169', this['map'].getView().getProjection());
```